src/main.cpp

```diff
@@ -43,24 +43,33 @@ void myMQTTCallBack(char* topic, byte* payload, unsigned int length) {
 }
 
 void setup() {
-  Serial.begin(115200);
+  //>>> hier wird "Serial" initialisiert und zum Schreiben von Debug-Ausgaben genutzt
+  Serial.begin(115200, SERIAL_8N1);
 
-  dbg.println("Start of Solar Inverter MQTT Gateway"); 
-  dbg.println("Starting BaseConfig");
+  Serial.printf("Start of Solar Inverter MQTT Gateway V%s\n", Release); 
+  Serial.println("Starting BaseConfig");
+
+  //>>> hier wird die Config eingelesen, u.a. die Rx/Tx-Pins für die Debug-Ausgaben
+  //>>> "BaseConfig" nutzt schon "dbg", das im Fall USE_WEBSERIAL noch garnicht initialisiert ist
   Config = new BaseConfig();
 
   #ifndef USE_WEBSERIAL
-    dbg.begin(115200, SERIAL_8N1, Config->GetSerialRx(), Config->GetSerialTx()); // RX, TX, zb.: 33, 32
-    dbg.println("");
-    dbg.println("ready");
+    //Serial.begin(115200, SERIAL_8N1, Config->GetSerialRx(), Config->GetSerialTx()); // RX, TX, zb.: 33, 32 >>> hier würde "Serial" an "BaseConfig" angepasst werden...
```

Comment:
den serialport kann man nach belieben umbiegen.
diese definition ist eine advanced option wenn man zb. eine mittels eines 2. ESP eine TransparentBridge anlegt um das echte SerialLog Remote zu lesen wenn der ESP selbst nicht starten will

Reply:
ok, dafür also die Konfigurationsmlöglichkeit.
Bleibt noch das Initialisierproblem:
- in commonlibs.h definierst Du dbg zu "Serial" oder zu "WebSerial"
- in main.cpp / setup() wird zuerst ein Objekt der Klasse "BaseConfig" angelegt
- das ist ja auch nötig, um die Seriell-Konfiguration einzulesen
- allerdings arbeitet "BaseConfig" selbst schon mit "dbg", d.h. entweder dem default-Serial oder dem noch nicht initialisierten WebSerial
- nachdem BaseConfig angelegt ist, wird WebSerial initialisiert (setBuffer() und begin()) bzw. Serial parametriert...
Ich habe mich mit WebSerial nicht auseinandergesetzt, vielleicht darf es ja schon vor begin() verwendet werden - zumindest wollte ich Dich auf diese Stelle aufmerksam machen.
Eine saubere Lösung (aus meiner eingeschränkten Sicht) habe ich leider auch nicht, BaseConfig ohne Debug-Ausgaben ist natürlich nicht gut.